atlasdb-config/src/test/java/com/palantir/atlasdb/factory/AtlasDbDialogueServiceProviderTest.java

```diff
@@ -0,0 +1,157 @@
+
+        AtlasDbDialogueServiceProvider provider = AtlasDbDialogueServiceProvider.create(
+                Refreshable.only(serverListConfig),
+                DIALOGUE_BASE_FACTORY,
+                USER_USER_AGENT);
+        conjureTimelockService = provider.getConjureTimelockService();
+    }
+
+
+    @Test
+    public void canMakeRequestsThroughDialogue() {
+        ConjureGetFreshTimestampsResponse response = makeTimestampsRequest();
+        assertThat(response.getInclusiveLower()).isEqualTo(58);
+        assertThat(response.getInclusiveUpper()).isEqualTo(70);
+    }
+
+    @Test
+    public void requestsAreIdentifiedWithTheUserProvidedUserAgent() {
+        makeTimestampsRequest();
+
+        server.verify(postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader("User-Agent", WireMock.containing(USER_USER_AGENT.primary().name())));
+        server.verify(postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader("User-Agent", WireMock.containing(USER_USER_AGENT.primary().version())));
+    }
+
+    @Test
+    public void atlasDbHttpClientVersionProvidedAsAnInformationalAgent() {
+        makeTimestampsRequest();
+
+        server.verify(postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader("User-Agent", WireMock.containing(
+                        String.format("%s/%s",
+                                AtlasDbRemotingConstants.ATLASDB_HTTP_CLIENT,
+                                AtlasDbRemotingConstants.CURRENT_CLIENT_PROTOCOL_VERSION.getProtocolVersionString()))));
+    }
+
+    @Test
+    public void resilientToRepeatedRedirects() {
+        server.stubFor(TIMESTAMP_MAPPING.willReturn(aResponse()
+                .withStatus(308)
+                .withHeader("Location", getUriForPort(serverPort))));
+
+        Instant start = Instant.now();
+        ExecutorService ex = PTExecutors.newSingleThreadExecutor(true);
+        ex.submit(this::scheduleServerRecoveryAfterFiveSeconds);
+
+        assertThatCode(this::makeTimestampsRequest).doesNotThrowAnyException();
+        assertThat(Instant.now())
+                .as("should recover in a second after things are good again")
+                .isBefore(start.plus(Duration.ofSeconds(11)));
+        ex.shutdown();
+    }
+
+    private void scheduleServerRecoveryAfterFiveSeconds() {
+        Uninterruptibles.sleepUninterruptibly(5, TimeUnit.SECONDS);
+        setupServerToGiveOutTimestamps();
+    }
+
+    private void setupServerToGiveOutTimestamps() {
+        server.stubFor(TIMESTAMP_MAPPING.willReturn(aResponse()
+                .withStatus(200)
+                .withHeader("Content-Type", "application/json")
+                .withBody("{\"inclusiveLower\": 58, \"inclusiveUpper\": 70}")));
+    }
+
+    private ConjureGetFreshTimestampsResponse makeTimestampsRequest() {
+        return conjureTimelockService.getFreshTimestamps(
+                AuthHeader.valueOf("Bearer unused"), CLIENT, ConjureGetFreshTimestampsRequest.of(10));
```

Comment:
Lets make this 13+ so that the response makes sense

Reply:
Yep, makes sense. (Good call on 13+, and not precisely 13, as well!)